### PR TITLE
feat(selected item validator): validate component based on selected i…

### PR DESCRIPTION
**Pull request by ericpitcher**
…tem.

### CODE REVIEW

File: src/bim-select.component.js [hunk 1]
```diff
@@ -69,7 +69,16 @@ exports.name = 'bimSelect';
...
  *   and the `model` property has the item in the `items` array as a value.
- *
+ * @param {Expression<Function>>} [validator]
+ *   This function allows the consumer to specify if the selected item is valid or not.
+ *   The function is optional and the selected item will always be valid if not supplied.
+ *   The validator function should return a boolean. It receives the selected item
+ *   as it's first parameter (which may be falsy if nothing is selected), it recieves the
+ *   component $element itself as the 2nd parameter. Returning true makes it valid, false
+ *   makes it invalid.
+ * @param {Expression<String>>} [invalidClasses]
```
[the remark this answers is not in the document]
Comment (tregusti) [reply src/bim-select.component.js]: One too many `>` here as well.

**Comment by tregusti** [comment]
Also, add yourself to the now non-existant `contributors` field.

https://docs.npmjs.com/files/package.json#people-fields-author-contributors

File: src/bim-select.template.html [hunk 2]
```diff
@@ -2,6 +2,9 @@
     <div class="input-group">
         <span ng-if="!$ctrl.shouldDisplayInput()" 
               class="bim-select-selected-item form-control"
+              ng-class="{
+                'is-clearable': $ctrl.isClearable() && !$ctrl.isDisabled()
```
[the remark this answers is not in the document]
Comment (ericpitcher) [reply src/bim-select.template.html]: It's in the component controller since before this feature

[the remark this answers is not in the document]
Comment (tregusti) [reply src/bim-select.template.html]: Oops.

**Comment by ericpitcher** [comment]
@tregusti removed $element argument

**Pull request merged**